server/graphql/invoice_line/src/mutations/inbound_shipment_line/line/zero_line_quantity.rs

```diff
@@ -0,0 +1,139 @@
+use super::{BatchIsReserved, InvoiceWasCreatedAfterStore};
+use async_graphql::*;
+use graphql_core::standard_graphql_error::{validate_auth, StandardGraphqlError};
+use graphql_core::{
+    simple_generic_errors::{CannotEditInvoice, ForeignKey, ForeignKeyError, RecordNotFound},
+    ContextExt,
+};
+
+use graphql_types::types::InvoiceLineNode;
+use repository::InvoiceLine;
+use service::auth::{Resource, ResourceAccessRequest};
+use service::invoice_line::inbound_shipment_line::{
+    ZeroInboundShipmentLineQuantity as ServiceInput,
+    ZeroInboundShipmentLineQuantityError as ServiceError,
+};
+
+#[derive(InputObject)]
+#[graphql(name = "ZeroInboundShipmentLineQuantityInput")]
+pub struct ZeroInboundShipmentLineQuantityInput {
+    pub id: String,
+}
+
+#[derive(SimpleObject)]
+#[graphql(name = "ZeroInboundShipmentLineQuantityError")]
+pub struct ZeroInboundShipmentLineQuantityError {
+    pub error: ZeroInboundShipmentLineQuantityErrorInterface,
+}
+
+#[derive(Union)]
+#[graphql(name = "ZeroInboundShipmentLineQuantityResponse")]
+pub enum ZeroInboundShipmentLineQuantityResponse {
+    Error(ZeroInboundShipmentLineQuantityError),
+    Response(InvoiceLineNode),
+}
+
+pub fn zero_inbound_shipment_line_quantity(
+    ctx: &Context<'_>,
+    store_id: &str,
+    input: ZeroInboundShipmentLineQuantityInput,
+) -> Result<ZeroInboundShipmentLineQuantityResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateInboundShipment,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    map_response(
+        service_provider
+            .invoice_line_service
+            .zero_inbound_shipment_line_quantity(&service_context, input.to_domain()),
+    )
+}
+
+#[derive(Interface)]
+#[graphql(name = "ZeroInboundShipmentLineQuantityErrorInterface")]
+#[graphql(field(name = "description", type = "&str"))]
+pub enum ZeroInboundShipmentLineQuantityErrorInterface {
+    RecordNotFound(RecordNotFound),
+    ForeignKeyError(ForeignKeyError),
+    CannotEditInvoice(CannotEditInvoice),
+    BatchIsReserved(BatchIsReserved),
+    InvoiceWasCreatedAfterStore(InvoiceWasCreatedAfterStore),
```

Comment:
Do we need these? e.g.:

- RecordNotFound(RecordNotFound) -> BadUserInput/ Internal error?
- ForeignKeyError(ForeignKeyError), -> internal error
- CannotEditInvoice(CannotEditInvoice), -> BadUserInput? (what is the reason here?)
- BatchIsReserved(BatchIsReserved), -> BadUserInput
- InvoiceWasCreatedAfterStore(InvoiceWasCreatedAfterStore), -> BadUserInput